index.d.ts

```diff
@@ -626,3 +626,66 @@ interface KVNamespace {
     cursor: string;
   }>;
 }
+
+interface DurableObjectEntries<T> {
+  [key: string]: T;
+}
+
+interface DurableObjectListOptions {
+  start?: string;
+  end?: string;
+  reverse?: boolean;
+  limit?: number;
+}
+
+interface DurableObjectOperator {
+  get<T = unknown>(key: string): Promise<T>;
+  get<T = unknown>(keys: Array<string>): Promise<Map<string, T>>;
+  put<T = unknown>(key: string, value: T): Promise<void>;
+  put<T = unknown>(entries: DurableObjectEntries<T>): Promise<void>;
+  delete(key: string): Promise<boolean>;
+  delete(keys: Array<string>): Promise<boolean>;
+  list<T = unknown>(options?: DurableObjectListOptions): Promise<Map<string, T>>;
+}
+
+interface DurableObjectTransaction extends DurableObjectOperator {
+  rollback(): void;
+}
+
+interface DurableObjectStorage extends DurableObjectOperator {
+  transaction<T = unknown>(
+    closure: (txn: DurableObjectStorage) => Promise<void>
+  ): Promise<Map<string, T>>;
+}
+
+interface DurableObjectState {
+  storage: DurableObjectStorage;
+}
+
+/**
+ * DurableObject is a class that defines a template for creating Durable Objects
+ */
+interface DurableObject {
+  fetch: (request: Request, init?: RequestInfo) => Promise<Response>;
+}
+
+/**
+ * DurableObjectStub is a client object used to send requests to a remote Durable Object
+ */
+interface DurableObjectStub {
+  name?: string;
+  id: DurableObjectId;
+  fetch: (request: Request, init?: RequestInfo) => Promise<Response>;
+}
+
+interface DurableObjectId {
```

Comment:
Can this just be a `string`? IDs will be used often and this seems like syntax sugar.

Reply:
I believe DurableObjectId is an object not a string

Reply:
Yeah it's an object.  Even if it was just a string under the covers, it'd be better to have a type to distinguish it.

Also has an (optional) name property fwiw.

Reply:
Added name property